library/core/src/num/f32.rs

```diff
@@ -357,38 +357,76 @@ impl f32 {
     pub const MANTISSA_DIGITS: u32 = 24;
 
     /// Approximate number of significant digits in base 10.
+    ///
+    /// This is the maximum <i>x</i> such that any decimal number with <i>x</i>
+    /// significant digits can be converted to `f32` and back without loss.
```

Comment:
I admit I've not paged discussion on https://github.com/rust-lang/rust/pull/89238 into cache, but since that annotates this with "A minimum number of significant digits in base 10." it feels somewhat surprising to have "maximum" here.

cc @workingjubilee 

(Perhaps the language is actually equivalent, I would need to actually thoroughly understand what the significance is to make a call there).

Reply:
If I understand correctly, `f32::DIGITS` is Rust's version of C's `FLT_DIG`, and `f64::DIGITS` is Rust's version of `DBL_DIG`. The [working draft N3054 of the C standard](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3054.pdf) describes those like this (with radix <i>b</i> = 2 in our case):

> number of decimal digits, <i>q</i>, such that any floating-point number with <i>q</i> decimal digits can be rounded into a floating-point number with <i>p</i> radix <i>b</i> digits and back again without change to the <i>q</i> decimal digits,

![image](https://user-images.githubusercontent.com/18604588/202571588-6c1b0f15-df6d-4cb1-abad-fcbbc187fad8.png)

I wrote "maximum" there because if `f32::DIGITS` is 6, then 6 is the maximum number of decimal digits that can be converted to `f32` losslessly. A 7-digit decimal converted to `f32` and back can be altered by the conversions.

Reply:
An example for `f32` where a 7-digit number has a lossy conversion:
```rust
let s1 = "8.589973e9";
let float: f32 = s1.parse().unwrap();
let s2 = format!("{float:.6e}");
assert_eq!(s1, s2);
```
panics with
```
assertion failed: `(left == right)`
  left: `"8.589973e9"`,
 right: `"8.589974e9"`
```